network/message.go

```diff
@@ -0,0 +1,142 @@
+package network
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/imdario/mergo"
+	"github.com/nknorg/nconnect/admin"
+	"github.com/nknorg/nconnect/config"
+	"github.com/nknorg/nkn-sdk-go"
+	"github.com/nknorg/nkngomobile"
+)
+
+// msgType constants
+const (
+	MT_NONE                = 0
+	JOIN_NETWORK           = 1
+	UPDATE_MY_INFO         = 2
+	GET_MY_INFO            = 3
+	GET_NODES_I_ACCEPT     = 4
+	GET_NODES_I_CAN_ACCESS = 5
+	LEAVE_NETWORK          = 6
+
+	NOTI_AUTHORIZED    = 7
+	NOTI_NEW_MEMBER    = 8
+	NOTI_SET_ACCEPT    = 9
+	NOTI_MEMBER_ONLINE = 10
+)
+
+type nodeInfo struct {
+	IP       string    `json:"ip"`
+	Netmask  string    `json:"netmask"`
+	Name     string    `json:"name"`
+	Address  string    `json:"address"`
+	LastSeen time.Time `json:"lastSeen"`
+}
+
+type networkInfo struct {
+	Domain  string `json:"domain"`
+	Gateway string `json:"gateway"`
+	DNS     string `json:"dns"`
+}
+
+type memberToManager struct {
+	MsgType int    `json:"msgType"`
+	Name    string `json:"name"`
+}
+
+type managerToMember struct {
+	MsgType     int          `json:"msgType"`
+	Err         error        `json:"err"`
+	NetworkInfo *networkInfo `json:"networkInfo"`
+	NodeInfo    []*nodeInfo  `json:"nodeInfo"`
+}
+
+func SendMsg(mc *admin.Client, address string, msg interface{}, waitResponse bool) (*managerToMember, error) {

```

Comment:
Codes in this function seem to be mostly duplicated (admin/client.go:RPCCall())

Reply:
Yes, I have tried to set up a common function to send nkn messages in both cases. but it both sides have some different logic inside.

Reply:
Yes, modified.